### ISSUE TITLE
Refactor SEI to use pure URI

### ISSUE THREAD

**Issue by NeonDaniel** [OpenVoiceOS/ovos-technical-manual#25]
The [Stream Extractor ID](https://openvoiceos.github.io/ovos-technical-manual/ocp_plugins/) prefix on media URIs looks like and has [previously been confused as being a URI scheme](https://github.com/OpenVoiceOS/ovos-ocp-audio-plugin/pull/41).

I would propose updating this spec to simply use a [URI scheme](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax) so we pass standard URIs around. This behavior is standard for linking content to a particular application or identifying the type of content ([Windows examples](https://learn.microsoft.com/en-us/windows/uwp/launch-resume/launch-default-app)).

To migrate to using URIs, we would simply need to parse the scheme:
```python
scheme, path = uri.split(':', 1))
if '//' in scheme:
    log_deprecation("Update plugin to pass URI without SEI prefix", "0.1.0")
    scheme = scheme.split('//')[0]
    uri = f"{scheme}//:{path}"
```

**Comment by JarbasAl** [comment]
this was a conscious decision to make it **different** from a URI, since a real uri comes after the SEI

today i would probably implement it differently, but when it started we needed compat with mycroft and could not easily introduce new fields etc

for now this needs to stay as many many skills in the wild depend on it, was one of the early features of OCP  (the purpose being moving the extracting to playback time, not search time, greatly improving search time)

if it changes in the future, its most likely to become a new field in message.data  (also allowing for a slow migration)

**Comment by NeonDaniel** [comment]
> for now this needs to stay as many many skills in the wild depend on it, was one of the early features of OCP (the purpose being moving the extracting to playback time, not search time, greatly improving search time)

I suggested parsing to maintain backwards-compat. so an old skill or plugin would have time to update to pass a regular URI.

> if it changes in the future, its most likely to become a new field in message.data (also allowing for a slow migration)

Why add a separate field when there's an existing spec to pass that information in the URI?

My reasoning is that if we're just passing a reference to some media, a URI is universal by definition and we should use established standards. Adding custom parsing to file references makes it less clear what is happening and harder to integrate with other modules on the bus. For example, if we have a `youtube://` URI, that can be passed directly to a client device like a phone to open the link in the user's preferred app.

**Comment by JarbasAl** [comment]
those are different concepts and not really equivalent, these are internal OCP ids, that identify how to get a real stream, those are not meant to make sense outside of OCP, and not mean to be universal resource identifiers. a SEI is tied to a specific plugin running in OCP

they dont identify a resource, they tell OCP "you must send this to a plugin before playback to get the URI", they are not valid uris and not meant to be passed to client devices like phones...

these are not file references, those are instructions how to find the real file when using the specific piece of code running in OCP, which is not expected to be available in clients by definition, this is internal and not meant for external consumption. SEIs are meant to be handled before anything makes it to the client, the client gets a real regular URI for playback  (a client usually being the audio subsystem, but can be any client app like hivemind)

in short, a SEI is only meant to exist between skill result and OCP receiving the result, once anything is asked to do the actual playback by OVOS then a real URI is used
